connection.go

```diff
@@ -82,11 +82,20 @@ func (mc *mysqlConn) handleParams() (err error) {
 }
 
 func (mc *mysqlConn) Begin() (driver.Tx, error) {
+	return mc.begin(false)
+}
+
+func (mc *mysqlConn) begin(readOnly bool) (driver.Tx, error) {
 	if mc.closed.IsSet() {
 		errLog.Print(ErrInvalidConn)
 		return nil, driver.ErrBadConn
 	}
-	err := mc.exec("START TRANSACTION")
+	var err error
+	if readOnly {
+		err = mc.exec("START TRANSACTION READ ONLY")
```

Comment:
It's likely better if the if-else just changes the query string, e.g. it should result in a bit less generated code.